src/ts/spatialnavigation/navigationgroup.ts

```diff
@@ -0,0 +1,141 @@
+import { Action, Direction } from './spatialnavigation';
+import { Container } from '../components/container';
+import { Component } from '../components/component';
+import { getElementInDirection } from './navigationalgorithm';
+import { getHtmlElementsFromComponents } from './gethtmlelementsfromcomponents';
+import { NodeEventSubscriber } from './nodeeventsubscriber';
+import { isSettingsPanel } from './typeguards';
+
+export type NavigationCallback = (direction: Direction, target: HTMLElement) => boolean;
+export type ActionCallback = (action: Action, target: HTMLElement) => boolean;
+
+export class NavigationGroup {
+  private activeElement?: HTMLElement;
+  private activeElementBeforeDisable?: HTMLElement;
+  private readonly components: Component<unknown>[];
+  private removeElementHoverEventListeners = () => {};
+  private readonly eventSubscriber: NodeEventSubscriber;
+
+  constructor(public readonly container: Container<unknown>, ...components: Component<unknown>[]) {
+    this.components = components;
+    this.eventSubscriber = new NodeEventSubscriber();
+  }
+
+  /**
+   * Adds onNavigation override
+   * @param direction {Direction}
+   * @param target {HTMLElement}
+   *
+   * @returns Boolean - true means it should stop propagation, and false means it should be handled by default handler
```

Comment:
Assigning such meaning to a callback return value seems a bit arbitrary. `return true;` or `return false;` has no implicit meaning in a user-implemented function.
Have you considered to instead pass a `preventDefault()` function instead?

Reply:
Agreed. Fixed in be6e4b57 - we're now using `preventDefault`.